src/app/bungie-icon/bungie-icon.component.ts

```diff
@@ -0,0 +1,17 @@
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'd2c-bungie-icon',
+  template: `
+    <img class="bungie-icon" [ngClass]="iconClass" [src]="iconUrl">
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class BungieIconComponent {
+
+  @Input() set icon(path: string) { this.iconUrl = `http://www.bungie.net${path}`; }
+  @Input() iconClass: string = 'icon';
+
+  public iconUrl: string;
+
+}
```

Comment:
Utility class that saved me a ton of repeated code. I stuck it in the base app for re-use elsewhere